lib/websocket.js

```diff
@@ -17,29 +17,109 @@ const OPCODE_SHORT = 0x81;
 const LEN_16_BIT = 126;
 const MAX_16_BIT = 65536;
 const LEN_64_BIT = 127;
+const MAX_64_BIT_PAYLOAD = Number.MAX_SAFE_INTEGER;
+const HEAD_LENGTH = 2;
+const EXTENDED_PAYLOAD_16_LENGTH = 2;
+const EXTENDED_PAYLOAD_64_LENGTH = 8;
 
-const calcOffset = (frame, length) => {
-  if (length < LEN_16_BIT) return [2, 6];
-  if (length === LEN_16_BIT) return [4, 8];
-  return [10, 14];
-};
+class Frame {
+  #frame = null;
+  #masked = null;
+  #dataOffset = null;
+  #mask = null;
+  #length = null;
 
-const parseFrame = (frame) => {
-  const length = frame[1] ^ 0x80;
-  const [maskOffset, dataOffset] = calcOffset(frame, length);
-  const mask = frame.subarray(maskOffset, maskOffset + MASK_LENGTH);
-  const data = frame.subarray(dataOffset);
-  return { mask, data };
-};
+  constructor(frame) {
+    this.#frame = frame;
+    const length = this.#frame[1] & 0x7f;
```

Comment:
Looks like `new Frame()` will return error. 
` Uncaught TypeError: Cannot read properties of undefined (reading '1')`
Is it ok here?

Reply:
Because it doesn't have to work that way, pass a reference to the WebSocket frame buffer to the constructor.

Reply:
Or use the method from